### PR TITLE
feat: split containers-kubernetes rules

### PR DESCRIPTION
**Pull request by rajatagarwal-ibm**
### Description

created 2 different rules for containers-kubernetes

One for api-type:cluster, and another one for api-type:management.

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by vburckhardt** [comment]
Hello @rajatagarwal-ibm  - does it end up in 2 different CBR rules? A quick scan through the code makes me think that we'll still end up with 1 CBR rules with 2 API types attached to it.

**Comment by rajatagarwal-ibm** [comment]
Thanks for the review @vburckhardt, as discussed I will have a look if 2 different keys can be created in `target_service_details_default` local variable OR if we can specify an array for both the rules in the "containers-kubernetes", which may require design review and further code changes with its implications.

**Comment by rajatagarwal-ibm** [comment]
Output: 

image

**Comment by vburckhardt** [comment]
@rajatagarwal-ibm - look at my latest commit. I refactored the code to make it more generic and shorter, so that we can easily extend it when ICD and other services surfaces multiple api types in future. I also added documentation.

**Review by vburckhardt — APPROVED** [review]

**Comment by rajatagarwal-ibm** [comment]
Tests are failing. Seems there are CBR rules and zones created which I need to delete first.

File: modules/fscloud/main.tf [hunk 1]
```diff
@@ -66,7 +66,10 @@ locals {
     "hs-crypto" : {
       "enforcement_mode" : "report"
     },
-    "containers-kubernetes" : {
+    "containers-kubernetes-management" : {
```
Comment (rajatagarwal-ibm) [inline modules/fscloud/main.tf:69]: Should we keep `containers-kubernetes` as well along with management and cluster?

Comment (vburckhardt) [reply modules/fscloud/main.tf:69]: I don't think so, this would result in duplicate rules

**Pull request merged**